types/node/crypto.d.ts

```diff
@@ -2309,11 +2309,11 @@ declare module 'crypto' {
          * If `encoding` is specified, a string is returned; otherwise a `Buffer` is
          * returned.
          * @since v0.11.14
-         * @param encoding The `encoding` of the return value.
+         * @param [encoding] The `encoding` of the return value.
          * @param [format='uncompressed']
          * @return The EC Diffie-Hellman public key in the specified `encoding` and `format`.
          */
-        getPublicKey(): Buffer;
+        getPublicKey(encoding?: null, format?: ECDHKeyFormat): Buffer;
```

Comment:
In practice it appears `encoding` can be any falsy value (0, false, etc...) but those seem like bad practice to encourage.

`undefined` is implied via optional `?` so tslint does not like it specified explicitly, so `null` is what I was left with.

Reply:
I'm thinking it'd actually be better to have `encoding?: undefined` only; [the docs](https://nodejs.org/api/crypto.html#ecdhgetpublickeyencoding-format) are worded as:

> If encoding is specified, a string is returned; otherwise a Buffer is returned.

Which inversely would imply that "A Buffer is returned if encoding is not specified", which to me sounds like not providing the parameter, which means undefined.

Reply:
If a falsy value works, then I can only imagine that crypto has JS code that just checks `if (!encoding)`. I don't know if it's worth trying so hard as to exactly emulate a not check.

Reply:
Ah, okay.